pkg/manifestreader/manifestloader.go

```diff
@@ -59,18 +59,24 @@ func (f *manifestLoader) ManifestReader(reader io.Reader, args []string) (Manife
 		EnforceNamespace: enforceNamespace,
 	}
 
+	return mReader(path, reader, readerOptions), nil
+}
+
+// mReader returns the ManifestReader based in the input args
+func mReader(path string, reader io.Reader, readerOptions ReaderOptions) ManifestReader {
 	var mReader ManifestReader
-	if len(args) == 0 {
+	// Read from stdin if "-" is specified, similar to kubectl
+	if path == "-" {
```

Comment:
This seems to change the current UX for the cli. I think this is something we want to avoid.